sycl/test-e2e/InvokeSimd/Spec/simd_mask_merge.cpp

```diff
@@ -0,0 +1,117 @@
+// TODO: enable on Windows once driver is ready
+// REQUIRES: gpu && linux
+// UNSUPPORTED: cuda || hip
+//
+// Check that full compilation works:
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+//
+// VISALTO enable run
+// RUN: env IGC_VISALTO=63 IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+
+#include <sycl/detail/boost/mp11.hpp>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/oneapi/experimental/invoke_simd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <functional>
+#include <iostream>
+#include <type_traits>
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+namespace esimd = sycl::ext::intel::esimd;
+constexpr int VL = 16;
+
+[[intel::device_indirectly_callable]]
+simd<float, VL>
+SIMD_CALLEE(simd<float, VL> va,
+             simd_mask<float, VL> mask) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> ret;
+  esimd::simd_mask<VL> emask;
+  for(int i = 0; i < VL; i++)
+    emask[i] = static_cast<bool>(mask[i]);
+  ret.merge(va, !emask);
+  return ret;
+}
+
+int main() {
+  sycl::queue q{gpu_selector_v};
```

Comment:
```suggestion
  sycl::queue q;
```

because we already have `GPU_RUN_PLACEHOLDER`.

Reply:
will fix, thanks